### ISSUE TITLE
[TodoExample] Tapping Todo list items causes MobXException

### ISSUE THREAD

**Issue by jgroman** [mobxjs/mobx.dart#106]
Hi,

tapping Todo list items in TodoExample currently causes MobXException "Side effects like changing state are not allowed at this point."
I rebuilt generated file _todos.g.dart_ with `flutter packages pub run build_runner build` but the problem remains.

```
I/flutter (10412): ══╡ EXCEPTION CAUGHT BY GESTURE ╞═══════════════════════════════════════════════════════════════════
I/flutter (10412): The following MobXException was thrown while handling a gesture:
I/flutter (10412): Instance of 'MobXException'
I/flutter (10412):
I/flutter (10412): When the exception was thrown, this was the stack:
I/flutter (10412): #0      ReactiveContext.checkIfStateModificationsAreAllowed (file:///C:/project/flutter/mobx.dart/mobx/lib/src/core/context.dart:145:11)
I/flutter (10412): #1      Todo.done= (file:///c:/project/flutter/mobx.dart/flutter_mobx/example/lib/todos/todos.g.dart:35:17)
I/flutter (10412): #2      TodoListView.build.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (file:///c:/project/flutter/mobx.dart/flutter_mobx/example/lib/todos/todo_widgets.dart:57:56)
I/flutter (10412): #3      CheckboxListTile.build.<anonymous closure> (package:flutter/src/material/checkbox_list_tile.dart:206:43)
I/flutter (10412): #4      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:513:14)
I/flutter (10412): #5      _InkResponseState.build.<anonymous closure> (package:flutter/src/material/ink_well.dart:568:30)
I/flutter (10412): #6      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:120:24)
I/flutter (10412): #7      TapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:242:9)
I/flutter (10412): #8      TapGestureRecognizer.acceptGesture (package:flutter/src/gestures/tap.dart:204:7)
I/flutter (10412): #9      GestureArenaManager.sweep (package:flutter/src/gestures/arena.dart:156:27)
I/flutter (10412): #10     _WidgetsFlutterBinding&BindingBase&GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:184:20)
I/flutter (10412): #11     _WidgetsFlutterBinding&BindingBase&GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:158:22)
I/flutter (10412): #12     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:138:7)
I/flutter (10412): #13     _WidgetsFlutterBinding&BindingBase&GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:101:7)
I/flutter (10412): #14     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:85:7)
I/flutter (10412): #18     _invoke1 (dart:ui/hooks.dart:203:10)
I/flutter (10412): #19     _dispatchPointerDataPacket (dart:ui/hooks.dart:138:5)
I/flutter (10412): (elided 3 frames from package dart:async)
I/flutter (10412):
I/flutter (10412): Handler: onTap
I/flutter (10412): Recognizer:
I/flutter (10412):   TapGestureRecognizer#43687(debugOwner: GestureDetector, state: ready, won arena, finalPosition:
I/flutter (10412):   Offset(129.1, 411.0), sent tap down)
I/flutter (10412): ════════════════════════════════════════════════════════════════════════════════════════════════════
```

**Comment by pavanpodila** [comment]
Ya this has to do with the newly introduced `enforceActions` check in #88

**Comment by pavanpodila** [comment]
fixed with #107
